lib/server/api.js

```diff
@@ -265,6 +265,46 @@ function generateSubapp(options) {
       }, next);
   });
 
+  /**
+   * Creates a SSH key for the specified :user
+   */
+  subapp.post('/datacenters/:dc/users/:user/sshkeys', function (req, res, next) {
+    if (!req.body.key) {
+      return res
+        .status(400)
+        .send({
+          message: 'must specify public key'
+        });
+    }
+
+    client
+      .createKey(req.param('user'), req.param('dc'), {
+        name: req.body.name,
+        key: req.body.key
+      })
+      .then(function (createdKey) {
+        return res
+          .status(201)
+          .send({
+            sshKey: {
+              name: createdKey.name,
+              key: createdKey.key,
+              fingerprint: createdKey.fingerprint
+            }
+          });
+      })
+      .catch(function (error) {
+        return error && error.statusCode === 409;
+      }, function (error) {
```

Comment:
The first function is utilizing the <code>predicate</code> argument and is acting as a filter so that the second function, the true error handler, is only run if we get a 409 from Joyent's cloudapi client. It returns a 409 when the key you are uploading is invalid or already created.